internal/host/plugin/host_catalog_secret.go

```diff
@@ -90,41 +88,10 @@ func (c *HostCatalogSecret) decrypt(ctx context.Context, cipher wrapping.Wrapper
 	return nil
 }
 
-// TODO(ICU-2835): this is necessary as update logic requires a static
-// public_id or private_id field. Once post-lookup updates can
-// support alternate primary key fields, we can port this to native
-// Update and upserts w/OnConflict.
-func (c *HostCatalogSecret) upsertQuery() (query string, queryValues []interface{}) {
-	query = upsertHostCatalogSecretQuery
-	queryValues = []interface{}{
-		sql.Named("catalog_id", c.CatalogId),
-		sql.Named("secret", c.CtSecret),
-		sql.Named("key_id", c.KeyId),
+func (c *HostCatalogSecret) oplog(op oplog.OpType) oplog.Metadata {
```

Comment:
I don't think you need to define oplog() at all, do you?

Reply:
Is there an idiomatic way to add oplog messages in write queries without it?

Static catalogs have a similar pattern ([reference](https://github.com/hashicorp/boundary/blob/0a83e21aa82004f72b0c52d38166c099594db2ce/internal/host/static/repository_host_catalog.go#L146-L173)), but it's just a separate function versus a method. This more follows the pattern we use in the plugin-based host catalogs (ie: [`oplog`](https://github.com/hashicorp/boundary/blob/0a83e21aa82004f72b0c52d38166c099594db2ce/internal/host/plugin/host_catalog.go#L122-L132) for `HostCatalog` itself).

Reply:
Oh right, you know what, I do remember now - I was looking at adding this into the credential maintenance job, but it needs to be removed here for sure. I'll fix that before merging.